File: python/design_hash_set.py
class Bucket:
    def __init__(self):
        self.list = []

    def contains(self, element: int) -> bool:
        return element in self.list

    def add(self, element: int) -> None:
        self.list.append(element)

    def remove(self, element: int) -> None:
        if self.contains(element):
            self.list.remove(element)


class MyHashSet:
    def __init__(self):
        self._initial_size = 100
        self.size = 0
        self.buckets = [Bucket() for _ in range(self._initial_size)]

    def add(self, key: int) -> None:
        hashCode = key % len(self.buckets)
        if not self.buckets[hashCode].contains(key):
            self.buckets[hashCode].add(key)
            self.size += 1

    def remove(self, key: int) -> None:
        self.buckets[key % len(self.buckets)].remove(key)

    def contains(self, key: int) -> bool:
        return self.buckets[key % len(self.buckets)].contains(key)

File: src/DesignHashSet.java
import java.util.LinkedList;
import java.util.Queue;

public class DesignHashSet {
    class MyHashSet {

        private static final int INITIAL_SIZE = 100;
        private final Bucket[] buckets;
        private int size = 0;

        public MyHashSet() {
            this.buckets = new Bucket[INITIAL_SIZE];
            initializeBuckets();
        }

        public MyHashSet(int initialSize) {
            this.buckets = new Bucket[initialSize];
            initializeBuckets();
        }

        public void add(int element) {
            int hash = element % buckets.length;
            if (!buckets[hash].contains(element)) {
                buckets[hash].add(element);
                this.size++;
            }
        }

        public void remove(int element) {
            buckets[element % buckets.length].remove(element);
        }

        public boolean contains(int element) {
            return buckets[element % buckets.length].contains(element);
        }

        private void initializeBuckets() {
            for (int index = 0 ; index < buckets.length ; index++) {
                buckets[index] = new Bucket();
            }
        }

        private static class Bucket {
            Queue<Integer> list = new LinkedList<>();

            public boolean contains(int element) {
                return list.contains(element);
            }

            public void add(int element) {
                if (!this.contains(element)) {
                    list.add(element);
                }
            }

            public void remove(int element) {
                list.remove(element);
            }
        }
    }
}
